src/cleanfigure.m

```diff
@@ -864,25 +934,28 @@ function removeData(meta, handle, id_remove)
       return
   end
 
+  % Check whether this is a 3D plot
+  is3D = isAxis3D(meta.gca);
+
   % Extract the data from the current line handle.
   xData = get(handle, 'XData');
```

Comment:
I'm wondering whether what is the optimal way to express this.
Currently:
```matlab
xData = get(h, 'XData');
yData = get(h, 'YData');
if is3D
  zData = get(h, 'ZData');
end

xData(idx) = [];
yData(idx) = [];
if is3d
  zData(idx) = [];
end

set(h, 'XData', xData);
set(h, 'YData', yData);
if is3d
  set(h, 'ZData', zData);
end
```

versus

```matlab
removePoints(handle, 'XData', idx);
removePoints(handle, 'YData', idx);
if is3d
  removePoints(handle, 'ZData', idx);
end

function removePoints(h, propertyName, idx)
    % deletes the (X,Y,Z)Data at certain indices
    data = get(h, propertyName);
    data(idx) = [];
    set(h, propertyName, data);
end
```

The latter is shorter (and less duplication of code), but I wonder whether it is readable enough. What do you think?